mmv1/third_party/terraform/utils/test_utils.go

```diff
@@ -156,3 +163,81 @@ func checkDataSourceStateMatchesResourceStateWithIgnores(dataSourceName, resourc
 		return nil
 	}
 }
+
+// General test utils
+
+// MuxedProviders returns the correct test provider (between the sdk version or the framework version)
+func MuxedProviders(testName string) (func() tfprotov5.ProviderServer, error) {
+	ctx := context.Background()
+
+	providers := []func() tfprotov5.ProviderServer{
+		GetSDKProvider(testName).GRPCProvider, // sdk provider
+	}
+
+	muxServer, err := tf5muxserver.NewMuxServer(ctx, providers...)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return muxServer.ProviderServer, nil
+}
+
+func RandString(t *testing.T, length int) string {
+	if !isVcrEnabled() {
+		return acctest.RandString(length)
+	}
+	envPath := os.Getenv("VCR_PATH")
+	vcrMode := os.Getenv("VCR_MODE")
+	s, err := vcrSource(t, envPath, vcrMode)
+	if err != nil {
+		// At this point we haven't created any resources, so fail fast
+		t.Fatal(err)
+	}
+
+	r := rand.New(s.source)
+	result := make([]byte, length)
+	set := "abcdefghijklmnopqrstuvwxyz012346789"
+	for i := 0; i < length; i++ {
+		result[i] = set[r.Intn(len(set))]
+	}
+	return string(result)
+}
+
+func RandInt(t *testing.T) int {
+	if !isVcrEnabled() {
+		return acctest.RandInt()
+	}
+	envPath := os.Getenv("VCR_PATH")
+	vcrMode := os.Getenv("VCR_MODE")
+	s, err := vcrSource(t, envPath, vcrMode)
+	if err != nil {
+		// At this point we haven't created any resources, so fail fast
+		t.Fatal(err)
+	}
+
+	return rand.New(s.source).Int()
+}
+
+// ProtoV5ProviderFactories returns a muxed ProviderServer that uses the provider code from this repo (SDK and plugin-framework).
+// Used to set ProtoV5ProviderFactories in a resource.TestStep within an acceptance test.
+func ProtoV5ProviderFactories(t *testing.T) map[string]func() (tfprotov5.ProviderServer, error) {
+	return map[string]func() (tfprotov5.ProviderServer, error){
+		"google": func() (tfprotov5.ProviderServer, error) {
+			provider, err := MuxedProviders(t.Name())
+			return provider(), err
+		},
+	}
+}
+
+// ProtoV5ProviderBetaFactories returns the same as ProtoV5ProviderFactories only the provider is mapped with
+// "google-beta" to ensure that registry examples use `google-beta` if the example is versioned as beta;
+// normal beta tests should continue to use ProtoV5ProviderFactories
+func ProtoV5ProviderBetaFactories(t *testing.T) map[string]func() (tfprotov5.ProviderServer, error) {
+	return map[string]func() (tfprotov5.ProviderServer, error){
+		"google-beta": func() (tfprotov5.ProviderServer, error) {
```

Comment:
We'll want to replicate the rules from https://github.com/GoogleCloudPlatform/magic-modules/blob/441886964febfd4bfc4465dd7ff80ecbe0c52310/mmv1/third_party/terraform/utils/provider_test.go.erb#L149-L152, using all versions MMv1 is aware of. I think the google-internal EAP/alpha provider tests depend on it.